File: javascript/6_kyu/replace_with_alphabet_position.js
// Replace With Alphabet Position
// https://www.codewars.com/kata/546f922b54af40e1e90001da/train/javascript

function alphabetPosition(text) {
  const letter = "abcdefghijklmnopqrstuvwxyz";
  var alpha_index = {};
  var ans = [];

  for (var i = 0; i < letter.length; i++) {
    alpha_index[letter[i]] = i + 1;
  }

  for (var i = 0; i < text.length; i++) {
    if ( text[i].toLowerCase() in alpha_index ) {
      ans.push(alpha_index[text[i].toLowerCase()]);
    }
  }

  return ans.join(" ");
}

console.log(alphabetPosition("The sunset sets at twelve o' clock."));
console.log(alphabetPosition("The narwhal bacons at midnight."));

File: python/6_kyu/replace_with_alphabet_position.py
# Replace With Alphabet Position
# https://www.codewars.com/kata/546f922b54af40e1e90001da/train/python

def alphabet_position(text):
  letters = "abcdefghijklmnopqrstuvwxyz"
  alpha = {}
  ans = []

  for i in range(0, len(letters)):
    alpha[letters[i]] = i + 1
  
  for i in text:
    if i.lower() in alpha:
      ans.append(str(alpha[i.lower()]))

  return " ".join(ans)

print(alphabet_position("The sunset sets at twelve o' clock."))
print(alphabet_position("The narwhal bacons at midnight."))